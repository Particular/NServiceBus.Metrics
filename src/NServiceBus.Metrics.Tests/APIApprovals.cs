namespace NServiceBus.Metrics.Tests
{
    using System;
    using System.IO;
    using System.Linq;
    using System.Runtime.CompilerServices;
    using ApprovalTests;
    using ApprovalTests.Reporters;
    using NUnit.Framework;
    using PublicApiGenerator;

    [TestFixture]
    public class APIApprovals
    {
        [Test]
        [MethodImpl(MethodImplOptions.NoInlining)]
        [UseReporter(typeof(DiffReporter))]
        public void ApproveAPI()
        {
            var publicApi = Filter(ApiGenerator.GeneratePublicApi(typeof(MetricsFeature).Assembly));
            Approvals.Verify(publicApi);
        }

        [Test]
        [MethodImpl(MethodImplOptions.NoInlining)]
        [UseReporter(typeof(DiffReporter))]
        public void ApproveMetrics()
        {
            var writer = new StringWriter();
            AllMetrics.Define(writer);
            Approvals.Verify(writer);
        }

        string Filter(string text)
        {
            return string.Join(Environment.NewLine, text.Split(new[]
            {
                Environment.NewLine
            }, StringSplitOptions.RemoveEmptyEntries)
                .Where(l => !l.StartsWith("[assembly: ReleaseDateAttribute("))
                .Where(l => !string.IsNullOrWhiteSpace(l))
                );
        }

    }
}