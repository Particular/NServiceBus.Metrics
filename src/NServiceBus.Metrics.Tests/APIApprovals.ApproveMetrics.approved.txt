{
  "Version":"2",
  "Timestamp":"1970-01-01T00:00:00.0000",
  "Meters":[{
      "Name":"# of message failures / sec",
      "Count":0,
      "MeanRate":0.00,
      "OneMinuteRate":0.00,
      "FiveMinuteRate":0.00,
      "FifteenMinuteRate":0.00,
      "Unit":"Messages",
      "RateUnit":"s"
    },{
      "Name":"# of messages pulled from the input queue / sec",
      "Count":0,
      "MeanRate":0.00,
      "OneMinuteRate":0.00,
      "FiveMinuteRate":0.00,
      "FifteenMinuteRate":0.00,
      "Unit":"Messages",
      "RateUnit":"s"
    },{
      "Name":"# of messages successfully processed / sec",
      "Count":0,
      "MeanRate":0.00,
      "OneMinuteRate":0.00,
      "FiveMinuteRate":0.00,
      "FifteenMinuteRate":0.00,
      "Unit":"Messages",
      "RateUnit":"s"
    }],
  "Timers":[{
      "Name":"Critical Time",
      "Count":0,
      "ActiveSessions":0,
      "TotalTime":0,
      "Rate":{
          "MeanRate":0.00,
          "OneMinuteRate":0.00,
          "FiveMinuteRate":0.00,
          "FifteenMinuteRate":0.00
        },
      "Histogram":{
          "LastValue":0.00,
          "Min":0.00,
          "Mean":0.00,
          "StdDev":0.00,
          "Median":0.00,
          "Percentile75":0.00,
          "Percentile95":0.00,
          "Percentile98":0.00,
          "Percentile99":0.00,
          "Percentile999":0.00,
          "SampleSize":0
        },
      "Unit":"Messages",
      "RateUnit":"s",
      "DurationUnit":"ms"
    },{
      "Name":"Processing Time",
      "Count":0,
      "ActiveSessions":0,
      "TotalTime":0,
      "Rate":{
          "MeanRate":0.00,
          "OneMinuteRate":0.00,
          "FiveMinuteRate":0.00,
          "FifteenMinuteRate":0.00
        },
      "Histogram":{
          "LastValue":0.00,
          "Min":0.00,
          "Mean":0.00,
          "StdDev":0.00,
          "Median":0.00,
          "Percentile75":0.00,
          "Percentile95":0.00,
          "Percentile98":0.00,
          "Percentile99":0.00,
          "Percentile999":0.00,
          "SampleSize":0
        },
      "Unit":"Messages",
      "RateUnit":"s",
      "DurationUnit":"ms"
    }]
}
